app/integration/project-generator/src/main/java/io/syndesis/integration/project/generator/ProjectGenerator.java

```diff
@@ -82,6 +98,94 @@ public InputStream generate(IntegrationDeployment deployment) throws IOException
+                        .map(WithConfiguredProperties.class::cast)
+                        .map(WithConfiguredProperties::getConfiguredProperties)
+                        .flatMap(map -> map.entrySet().stream())
+                        .filter(Predicates.or(connector::isSecret, action::isSecret))
+                        .distinct()
+                        .forEach(
+                            e -> {
+                                String key = String.format("%s-%s.%s", componentScheme, index, e.getKey());
+                                String val = resourceManager.decrypt(e.getValue());
+
+                                properties.put(key, val);
+                            }
+                        );
+                } else {
+                    // The component scheme is defined as camel connector prefix
+                    // for 'old' style connectors.
+                    final String componentScheme = descriptor.getCamelConnectorPrefix();
+
+                    // endpoint secrets
+                    Stream.of(connector, connection, step)
+                        .filter(WithConfiguredProperties.class::isInstance)
+                        .map(WithConfiguredProperties.class::cast)
+                        .map(WithConfiguredProperties::getConfiguredProperties)
+                        .flatMap(map -> map.entrySet().stream())
+                        .filter(Predicates.or(connector::isEndpointProperty, action::isEndpointProperty))
+                        .filter(Predicates.or(connector::isSecret, action::isSecret))
+                        .distinct()
+                        .forEach(
+                            e -> {
+                                String key = String.format("%s-%s.%s", componentScheme, index, e.getKey());
+                                String val = resourceManager.decrypt(e.getValue());
+
+                                properties.put(key, val);
+                            }
+                        );
+
+                    // Component properties triggers connectors aliasing so we
+                    // can have multiple instances of the same connectors
+                    Stream.of(connector, connection, step)
+                        .filter(WithConfiguredProperties.class::isInstance)
+                        .map(WithConfiguredProperties.class::cast)
+                        .map(WithConfiguredProperties::getConfiguredProperties)
+                        .flatMap(map -> map.entrySet().stream())
+                        .filter(Predicates.or(connector::isComponentProperty, action::isComponentProperty))
+                        .distinct()
+                        .forEach(
+                            e -> {
+                                String key = String.format("%s.configurations.%s-%s.%s", componentScheme, componentScheme, index, e.getKey());
+                                String val = resourceManager.decrypt(e.getValue());
```

Comment:
dito wrt error handling. Maybe this inner decrypt part at least could be moved in a common method ? (so that e.g. the property key generation is guaranteed to be the same for all variants ?)

Reply:
Let me see what I can do.